tests/framework-tests/core/js/DataSourceTests.js

```diff
@@ -0,0 +1,191 @@
+            setTimeout(function () {
+                var response;
+                if (type === "set") {
+                    response = that[type](directModel, {ID: ++count}, {duration: duration});
+                } else {
+                    response = that[type](directModel, {
+                        duration: duration,
+                        ID: ++count
+                    });
+                }
+                response.then(function (val) {
+                    that.fireRecord[val.requestType].push(val.ID);
+                });
+            }, delay);
+        });
+    };
+
+    // s = short delay
+    // l = long delay
+    fluid.tests.requestRuns = {
+        ss: [0, 50, 55],
+        sl: [0, 50, 200],
+        ls: [0, 150, 200],
+        ll: [0, 150, 300],
+        sss: [0, 50, 55, 75],
+        lss: [0, 150, 200, 220],
+        ssl: [0, 50, 75, 200],
+        ssss: [0, 50, 55, 75, 80],
+        lsss: [0, 150, 200, 220, 250],
+        sssl: [0, 50, 60, 75, 200]
+    };
+
+    fluid.tests.expected100ms = {
+        ss: [1, 3],
+        sl: [1, 2, 3],
+        ls: [1, 2, 3],
+        ll: [1, 2, 3],
+        sss: [1, 4],
+        lss: [1, 2, 4],
+        ssl: [1, 3, 4],
+        ssss: [1, 5],
+        lsss: [1, 2, 5],
+        sssl: [1, 4, 5]
+    };
+
+    fluid.tests.assertRequest = function (requestType, setName, delays, expected, delayBuffer) {
+        // Time in milliseconds to add to the assertion delay, to buffer against setTimeout impressions.
+        // Because multiple instances run simultaneously this number may need to be increased to take into
+        // account extra delays related to the thread blocking.
+        delayBuffer = delayBuffer || 300;
+        var promise = fluid.promise();
+        var that = fluid.tests.queuedDataSource();
+        var assertionDelay = delays[delays.length - 1] + delayBuffer; // time to wait to assert the requests.
+
+        fluid.tests.invokeRequestWithDelay(that, requestType, delays, 100, {path: "value"});
+
+        setTimeout(function () {
+            jqUnit.assertDeepEq("The " + requestType  + " requests from set '" + setName +  "' should have been queued correctly.", expected, that.fireRecord[requestType]);
+            promise.resolve(that);
+        }, assertionDelay);
+        return promise;
+    };
+
+    fluid.tests.assertDelayedRequests = function (requestType, delaySet, expectedSet, delayBuffer) {
+        var count = 0;
+        var promise = fluid.promise();
+        var numSets = fluid.keys(delaySet).length;
+        fluid.each(delaySet, function (delay, set) {
+            var response = fluid.tests.assertRequest(requestType, set, delay, expectedSet[set], delayBuffer);
+            response.then(function () {
+                count++;
+                if (count >= numSets) {
+                    promise.resolve();
+                }
+            });
+        });
+        return promise;
+    };
+
+    jqUnit.asyncTest("Queued DataSource - get", function () {
```

Comment:
@jobara  - I mean to iterate over fluid.tests.queuedDataSource.requestRuns and for each entry there, to push a "function returning promise" into an array. The body of the function will perform the assertion with the appropriate options, and then resolve once the response has resolved. You can then drive this iteration from a declarative structure, e.g. containing elements such as {
   requestType: "get",
   source: fluid.tests.queuedDataSource.expected100ms
} 

etc - is this any clearer?